### PR TITLE
SDL_openslES.c: Detect float support in runtime

### LINKED ISSUE #5358: Android, OpenSL-ES: Use dynamic detection of SDK for Float output
Hello!

I had investigated for the reason of the long latency at me, and I found that you didn't use the dynamic detection of the output supported: https://github.com/libsdl-org/SDL/blob/a5c610b0a3857d3138f3f3da1f6dc3172c5ea4a8/src/audio/openslES/SDL_openslES.c#L417-L434

You can easily retrieve the SDK number and select the way dynamically: (on modern devices, allow floats to be used easily, and on old devices use PCM16 as fallback):
Here is my example of how I do that: https://github.com/Wohlstand/ADLMIDI-Player-Java/blob/master/ADLMIDI-Player/src/main/cpp/adlmidi-jni.c#L116-L172

And the code to retrieve the SDK version:
```c
#include <sys/system_properties.h>

/* ... */

long sdk_ver = 0;
char sdk_ver_str[92] = {'\0'}; /* Better just use 92 constant than PROPERTY_VALUE_MAX that locates at somewhere around */
__system_property_get("ro.build.version.sdk", sdk_ver_str);
sdk_ver = SDL_atoi(sdk_ver_str);

/* ... */

if(sdk_ver >= 21)
{
    /* Use any sample type including floats */
}
else
{
    /* Fallback to PCM16 */
}
```

### PR DESCRIPTION
**Pull request by Wohlstand**
Allow using of the float output type on newer Android devices, but keep PCM16 output on older

## Description
Here is possible to detect the support for the float type by dynamical retrieving of the SDK version. So, let's use this than just force everybody to use the limitation of old devices.

## Existing Issue(s)
Closes #5358

### CODE REVIEW

File: src/audio/openslES/SDL_openslES.c [hunk 1]
```diff
@@ -409,29 +410,34 @@ openslES_CreatePCMPlayer(_THIS)
     SLDataFormat_PCM format_pcm;
     SLresult result;
     int i;
+    long sdk_ver = 0;
+    char sdk_ver_str[92] = {'\0'};
+
+    __system_property_get("ro.build.version.sdk", sdk_ver_str);
+    sdk_ver = SDL_atoi(sdk_ver_str);
```
Comment (slouken) [inline src/audio/openslES/SDL_openslES.c:417]: You can use SDL_GetAndroidSDKVersion() here

Comment (Wohlstand) [reply src/audio/openslES/SDL_openslES.c:417]: Didn't knew about that call... Thanks

**Comment by slouken** [comment]
It looks like this patch will explicitly exclude floating point formats. How does this improve latency?

**Comment by slouken** [comment]
Your patch doesn't enable floating point support. :)
Are you saying that on your device AAudio has more latency than OpenSL-ES? Or that enabling float support on OpenSL-ES improves latency to match AAudio?

**Comment by Wohlstand** [comment]
> Your patch doesn't enable floating-point support. :)

Right, I had checked some at my code, and to enable floating-point support, needed to initialize the pcm_ex format instead of normal one:
```
   if(sdk_ver >= 21)
    {
        /* for Android 21+*/
        format_pcm_ex.formatType = SL_ANDROID_DATAFORMAT_PCM_EX;
        format_pcm_ex.numChannels = 2;
        format_pcm_ex.sampleRate = SL_SAMPLINGRATE_44_1;
        format_pcm_ex.bitsPerSample = SL_PCMSAMPLEFORMAT_FIXED_32;
        format_pcm_ex.containerSize = SL_PCMSAMPLEFORMAT_FIXED_32;
        format_pcm_ex.channelMask = SL_SPEAKER_FRONT_LEFT | SL_SPEAKER_FRONT_RIGHT;
        format_pcm_ex.endianness = SL_BYTEORDER_LITTLEENDIAN;
        format_pcm_ex.representation = SL_ANDROID_PCM_REPRESENTATION_FLOAT;
        audioSrc.pLocator = &locBufQ;
        audioSrc.pFormat = &format_pcm_ex;
    }
    else
    {
        /* for Android <21 */
        format_pcm.formatType = SL_DATAFORMAT_PCM;
        format_pcm.numChannels = 2;
        format_pcm.samplesPerSec = SL_SAMPLINGRATE_44_1;
        format_pcm.bitsPerSample = SL_PCMSAMPLEFORMAT_FIXED_16;
        format_pcm.containerSize = SL_PCMSAMPLEFORMAT_FIXED_16;
        format_pcm.channelMask = SL_SPEAKER_FRONT_LEFT | SL_SPEAKER_FRONT_RIGHT;
        format_pcm.endianness = SL_BYTEORDER_LITTLEENDIAN;
        audioSrc.pLocator = &locBufQ;
        audioSrc.pFormat = &format_pcm;
    }
```

When I spoke about latency, I didn't mean it's the main topic of this issue. This issue is about floating-point support at OpenSL-ES generally, out of my latency case. The latency problem can be discussed separately. I mentioned it because I found another problem while I investigated one. And this issue is about the floating-point problem, not about latency.

**Comment by Wohlstand** [comment]
Okay, now the code really initializes float output :)

**Comment by slouken** [comment]
Looks good, thanks!

**Pull request merged**